### PR TITLE
Referrals Gallery Update - referrals Prop to referralLabels

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request updates the `ReferralsGallery` `referrals` prop to be `referralLabels` -- from a list of user objects containing a `displayName` (which makes assumptions about the data source), to an agnostic list of label Strings. 

https://github.com/DoSomething/phoenix-next/pull/2207/commits/4d64de0ddb1681661a8e7fdc43ae4260728b55c1 addresses a phantom Signup Referrals Gallery that got left in via merge conflict resolution in #2205 

### How should this be reviewed?
👀 No visual changes!

### Any background context you want to provide?
This addresses feedback in https://github.com/DoSomething/phoenix-next/pull/2202#discussion_r436056696

### Relevant tickets

References [Pivotal #172748788](https://www.pivotaltracker.com/story/show/172748788).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js [hunk 1]
```diff
@@ -17,8 +17,6 @@ const ReferFriendsTab = () => (
         link="https://dosomething.org/us/campaigns/senior-homies"
         fullWidth
       />
-
-      <SignupReferralsGallery />
```
Comment (katiecrane) [inline resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js:21]: Why is this removed here?

Comment (mendelB) [reply resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js:21]: That one's a phantom Signup Referrals Gallery that got left in via merge conflict resolution in #2205.

Comment (katiecrane) [reply resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js:21]: Ooooo I see the one below now!
